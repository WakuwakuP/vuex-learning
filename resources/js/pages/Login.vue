<template>
  <div class="container--small">
    <ul class="tab">
      <li
        class="tab__item tab__item--login"
        :class="{'tab__item--active': tab === 'LoginPanel' }"
        @click="tab = 'LoginPanel'"
      >Login</li>
      <li
        class="tab__item tab__item--register"
        :class="{'tab__item--active': tab === 'RegisterPanel' }"
        @click="tab = 'RegisterPanel'"
      >Register</li>
    </ul>
    <transition name="panel" mode="out-in">
      <component v-bind:is="tab"></component>
    </transition>
  </div>
</template>

<script>
import { mapState } from "vuex";
import LoginPanel from "../components/LoginPanel";
import RegisterPanel from "../components/RegisterPanel";

export default {
  data() {
    return {
      tab: "LoginPanel"
    };
  },
  components: {
    LoginPanel,
    RegisterPanel
  },
  methods: {
    clearError() {
      this.$store.commit("auth/setLoginErrorMessages", null);
      this.$store.commit("auth/setRegisterErrorMessages", null);
    }
  },
  created() {
    this.clearError();
  },
  computed: {
    ...mapState({
      apiStatus: state => state.auth.apiStatus
    })
  }
};
</script>
