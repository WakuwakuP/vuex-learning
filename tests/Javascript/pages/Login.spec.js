import { createLocalVue, shallowMount, mount } from '@vue/test-utils';
import Vuex from 'vuex';

import Login from '@/pages/Login';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Login Lifecycle', () => {
  let store;
  let clearError;
  beforeEach(() => {
    store = new Vuex.Store({});
    clearError = jest.fn();
  });

  it('test', () => {
    shallowMount(Login, { store, localVue, methods: { clearError } });
    expect(clearError).toHaveBeenCalled();
  });
});

describe('Login Props/VuexState', () => {
  let store;
  let clearError;
  beforeEach(() => {
    const auth = {
      state: {
        loginErrorMessages: null,
        registerErrorMessages: null,
      },
    };
    store = new Vuex.Store({
      modules: {
        auth,
      },
    });
    clearError = jest.fn();
  });
  it('Login', () => {
    const wrapper = mount(Login, { store, localVue, methods: { clearError } });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Register', () => {
    const wrapper = mount(Login, {
      store,
      localVue,
      methods: {
        clearError,
      },
      data: () => ({
        tab: 'RegisterPanel',
      }),
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('User Interaction', () => {
  let store;
  let clearError;
  beforeEach(() => {
    const auth = {
      state: {
        loginErrorMessages: null,
        registerErrorMessages: null,
      },
    };
    store = new Vuex.Store({
      modules: {
        auth,
      },
    });
    clearError = jest.fn();
  });

  it('Click Login and Register', () => {
    const wrapper = mount(Login, { store, localVue, methods: { clearError } });

    wrapper.find('.tab__item--register').trigger('click');
    expect(wrapper.vm.tab).toBe('RegisterPanel');

    wrapper.find('.tab__item--login').trigger('click');
    expect(wrapper.vm.tab).toBe('LoginPanel');
  });
});
