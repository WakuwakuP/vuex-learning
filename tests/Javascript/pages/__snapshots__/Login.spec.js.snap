// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Props/VuexState Login 1`] = `
<div
  class="container--small"
>
  <ul
    class="tab"
  >
    <li
      class="tab__item tab__item--login tab__item--active"
    >
      Login
    </li>
     
    <li
      class="tab__item tab__item--register"
    >
      Register
    </li>
  </ul>
   
  <div
    class="panel"
    mode="out-in"
    name="panel"
  >
    <form
      class="form"
    >
      <!---->
       
      <label
        for="login-email"
      >
        Email
      </label>
       
      <input
        class="form__item"
        id="login-email"
        type="text"
      />
       
      <label
        for="login-password"
      >
        Password
      </label>
       
      <input
        class="form__item"
        id="login-password"
        type="password"
      />
       
      <div
        class="form__button"
      >
        <button
          class="button button--inverse"
          id="login-submit"
          type="submit"
        >
          login
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Login Props/VuexState Register 1`] = `
<div
  class="container--small"
>
  <ul
    class="tab"
  >
    <li
      class="tab__item tab__item--login"
    >
      Login
    </li>
     
    <li
      class="tab__item tab__item--register tab__item--active"
    >
      Register
    </li>
  </ul>
   
  <div
    class="panel"
    mode="out-in"
    name="panel"
  >
    <form
      class="form"
    >
      <!---->
       
      <label
        for="username"
      >
        Name
      </label>
       
      <input
        class="form__item"
        id="username"
        type="text"
      />
       
      <label
        for="email"
      >
        Email
      </label>
       
      <input
        class="form__item"
        id="email"
        type="text"
      />
       
      <label
        for="password"
      >
        Password
      </label>
       
      <input
        class="form__item"
        id="password"
        type="password"
      />
       
      <label
        for="password-confirmation"
      >
        Password (confirm)
      </label>
       
      <input
        class="form__item"
        id="password-confirmation"
        type="password"
      />
       
      <div
        class="form__button"
      >
        <button
          class="button button--inverse"
          type="submit"
        >
          register
        </button>
      </div>
    </form>
  </div>
</div>
`;
